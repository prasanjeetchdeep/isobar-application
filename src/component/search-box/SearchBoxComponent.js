import React, { useState } from "react";
import "./SearchBoxComponent.modules.scss";
import CancelIcon from "../../assets/images/Cancel.svg";
import SearchIcon from "../../assets/images/Search.svg";

const SearchBoxComponent = ({ onSearch }) => {
    const [search, setSearch] = useState("");
    const handleChange = search => {
        setSearch(search);
        onSearch(search);
    };

    return (
        <div className="flexbox search-container">
            <img
                className="search-icon"
                src={SearchIcon}
                alt="Search"
            />
            <input
                className="search-box input-font"
                type="text"
                value={search}
                placeholder="What do you want to learn?"
                onChange={e => handleChange(e.target.value)} />
            {search && <img
                src={CancelIcon}
                alt="Cancel"
                className="cancel-icon"
                onClick={() => handleChange("")} />}
        </div>
    );
};

export default SearchBoxComponent;