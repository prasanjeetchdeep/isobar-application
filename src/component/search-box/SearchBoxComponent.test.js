import React from 'react';
import { render } from '@testing-library/react';
import SearchBoxComponent from './SearchBoxComponent';

const noop = () => { };
it("should render the seaarch box component", () => {
  const { asFragment } = render(
    <SearchBoxComponent
      onSearch={noop} />);
  expect(asFragment()).toMatchSnapshot();
});
