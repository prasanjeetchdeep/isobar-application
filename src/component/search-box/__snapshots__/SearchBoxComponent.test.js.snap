// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the seaarch box component 1`] = `
<DocumentFragment>
  <div
    class="flexbox search-container"
  >
    <img
      alt="Search"
      class="search-icon"
      src="Search.svg"
    />
    <input
      class="search-box input-font"
      placeholder="What do you want to learn?"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
